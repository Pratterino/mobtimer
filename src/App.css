html, body {
  background: #20002c; /* fallback for old browsers */
  background: var(--background);
}

.app {
  color: var(--color);
  text-align: center;
}

h1, h2, h3, h4 {
  color: var(--header-color);
  font-weight: bold;
}

.hide {
  text-align: left;
  background: #222;
  color: #888;
  font-size: 11px;
  z-index: 999999999;
  position: sticky;
}

.pointer {
  cursor: pointer;
}
