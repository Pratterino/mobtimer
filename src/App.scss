html, body {
  background: #20002c; /* fallback for old browsers */
  background: sticky var(--background);
  height: 100%;
  min-height: 100vh;
}

.app {
  color: var(--color);
  text-align: center;
  min-height: 100vh;
}

h1, h2, h3, h4 {
  color: var(--header-color);
  font-weight: bold;
}

.hide {
  text-align: left;
  background: #222;
  color: #888;
  font-size: 11px;
  z-index: 999999999;
  position: sticky;
}

.pointer {
  cursor: pointer;
}

footer {
  box-sizing: border-box;
  position: absolute;
  bottom: 0;
  display: flex;
  width: 100%;
  padding: var(--base-padding);

  .footer__item {
    flex: 1;
    text-align: right;
    line-height: normal;

    p {
      text-align: right;
      flex-direction: column;
      align-self: flex-end;
      flex-wrap: wrap;
      align-content: space-between;
    }
  }
}
