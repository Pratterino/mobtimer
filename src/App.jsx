import React, {Component} from 'react';
import {bindActionCreators} from "redux";
import {connect} from "react-redux";

import Timer from './timer/Timer';
import {startTimer} from "./timer/timerActions"
import {usersSelector} from "./user/userReducer";
import Users from "./user/Users";
import Notifications from "./notifications/Notifications";
import Settings from "./settings/Settings";
import './App.css';

class App extends Component {
    startCountDown = () => {
        this.props.startTimer();
    };

    render() {
        return (
            <div className="app">
                {false && window.location.hostname === "localhost" &&
                <div className="hide">
                    <h3>ReduxState</h3>
                    <pre>{JSON.stringify(this.props.test, null, 2)}</pre>
                </div>
                }

                <Notifications/>

                <Settings/>

                <Users/>

                <Timer/>
            </div>
        );
    }
}

const mapStateToProps = state => ({
    test: state,
    timer: state.timer,
    users: usersSelector(state),
});

const mapDispatchToProps = dispatch => bindActionCreators({
    startTimer,
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(App);

