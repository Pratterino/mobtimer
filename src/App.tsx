import * as React from 'react';
import { useEffect, useState } from 'react';
import { bindActionCreators, Dispatch } from 'redux';
import { connect } from 'react-redux';
import { getParsedTimeRemaining } from './helper/TimerHelper';
import { fetchBackgroundImage } from './unsplashedActions';
import LazyLoad from 'react-lazy-load';
import Timer from './timer/Timer';
import Users from './user/Users';
import Notifications from './notifications/Notifications';
import Settings from './settings/Settings';
import SoundSelector from './sound/SoundSelector';
import ImageLoader from './helper/ImageLoader';
import './App.scss';

interface Props {
    timer: {
        leaderboard: Leaderboard
        metadata: {
            todaysSessionLength: number;
        };
    };
    settings: {
        devMode: boolean;
    };
    test?: object;
}

interface Leaderboard {
    [key: string]: number;
}

interface Unsplash {
    userLink: string;
    username: string;
    unsplashedLink: string;
    image: string;
    imageSmall: string;
}

function App({ timer, settings, test }: Props) {
    const [unsplash, setUnsplash] = useState<Unsplash | undefined>();

    useEffect(() => {
        fetchBackgroundImage().then(unsplashed => setUnsplash(unsplashed));
    }, []);

    const renderLeaderboard = () => {
        const sortedLeaderboard = sortProperties(timer.leaderboard);
        return (
            <ol>
                {sortedLeaderboard.map((item, i) => (
                    <li>
                        {item[0]} ({getParsedTimeRemaining(item[1])})
                    </li>
                ))}
            </ol>
        );
    };

    const sortProperties = (obj: Leaderboard) => {
        const sortable = [];
        for (const key in obj) {
            if (obj.hasOwnProperty(key)) {
                sortable.push([key, obj[key]]);
            }
        }
        sortable.sort((a, b) => +b[1] - +a[1]);
        return sortable;
    };

    return (
        <>
            <div className="app">
                {settings.devMode && false && (
                    <div className="hide">
                        <h3>ReduxState</h3>
                        <pre>{JSON.stringify(test, null, 2)}</pre>
                    </div>
                )}

                <Notifications/>
                <Settings/>
                <Users/>
                <Timer/>

                <footer>
                    <div className="footer__item">
                        <h4>Today's leaderboard</h4>
                        {renderLeaderboard()}
                    </div>

                    <div className="footer__item center">
                        <h4>Today's active mob time</h4>
                        <p>{getParsedTimeRemaining(timer.metadata.todaysSessionLength)}</p>
                    </div>

                    <div className="footer__item"/>
                    <div className="footer__item center">
                        <h4>Finish sound</h4>
                        <SoundSelector/>
                    </div>
                    <div className="footer__item">
                        {unsplash && (
                            <div className="unsplashed-credits">
                                Photo by <a href={unsplash.userLink}>{unsplash.username}</a> on{' '}
                                <a href={unsplash.unsplashedLink}>Unsplash</a>
                            </div>
                        )}
                    </div>
                </footer>
            </div>

            <div id="bg-image">
                {unsplash && (
                    <LazyLoad
                        width={100}
                        height={100}
                        debounce={false}
                        offsetVertical={500}
                    >
                        <ImageLoader src={unsplash.image} />
                    </LazyLoad>
                )}
            </div>
        </>
    );
}

const mapStateToProps = (state: any) => ({
    test: state,
    settings: state.settings,
    timer: state.timer,
});

const mapDispatchToProps = (dispatch: Dispatch) => bindActionCreators({}, dispatch);

export const unwrapped = App;
export default connect(mapStateToProps, mapDispatchToProps)(App);
