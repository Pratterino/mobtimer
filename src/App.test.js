import React from 'react';
import {unwrapped} from './App';

const renderComponent = (props) => (
    <unwrapped {...props}/>
);

describe('App', () => {
    test('render should match snapshot', () => {
        const wrapper = renderComponent();
        expect(wrapper).toMatchSnapshot();
    });
});
