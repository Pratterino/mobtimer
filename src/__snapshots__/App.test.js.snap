// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should match snapshot 1`] = `<unwrapped />`;
